packages/react-scripts/config/webpack.config.js

```diff
@@ -44,6 +44,14 @@ const shouldInlineRuntimeChunk = process.env.INLINE_RUNTIME_CHUNK !== 'false';
 // Check if TypeScript is setup
 const useTypeScript = fs.existsSync(paths.appTsConfig);
 
+let workboxConfig = options => options;
+// Get react-scripts configuration overrides if available
+try {
+  workboxConfig = require(paths.appWorkboxConfigJs);
```

Comment:
Should the default config be completely replaced or would it be more convenient to simply "append" the new config via the spread operator or `Object.assign`?

Reply:
The `workboxConfig` is a function, not an object. There is nothing to "append".

Reply:
Why is `workboxConfig` a function, after all, it's just returning an object? Why not just use a plain object, and in the js file you would be able to just export an object instead of a function that returns an object.

Reply:
I would be happy to change to just an object if the majority of people think it would be better, however, please take the following into consideration.

A function will allow devs to just change the things they need too instead of having to understand all the options available in the [GenerateSW Plugin](https://developers.google.com/web/tools/workbox/modules/workbox-webpack-plugin) (after all, `create-react-app` is suppose to make as many decisions as possible for devs, right?), while also allowing the flexibility to change the options completely.

If we changed to export an object, we could do a merge with the default options, however, this will mean the dev needs to understand how the merge will happen. Will it be a deep property merge or just a shallow merge of the top level properties? IMO better to leave the merge with defaults up to the dev.

E.g.

If dev wants to use all default options except add a path to `navigateFallbackBlacklist`:

```js
// instead of
module.exports = {
  clientsClaim: true,
  exclude: [/\.map$/, /asset-manifest\.json$/],
  importWorkboxFrom: 'cdn',
  navigateFallback: publicUrl + '/index.html',
  navigateFallbackBlacklist: [
    // Exclude URLs starting with /_, as they're likely an API call
    new RegExp('^/_'),
    // Exclude URLs containing a dot, as they're likely a resource in
    // public/ and not a SPA route
    new RegExp('/[^/]+\\.[^/]+$'),
    // My custom regex as well...
    new RegExp('[A|B]'),
  ],
}

// I can just do
module.exports = options => {
  options.navigateFallbackBlacklist.push(new RegExp('[A|B]'))
  return options
}
```

Also, if a dev wants to completely rewrite the options, they can simply do:

```js
module.exports = () => {
  return {
    // my own options
  }
}
```